test/e2e/storage/empty_dir_wrapper.go

```diff
@@ -214,25 +214,8 @@ func createGitServer(f *framework.Framework) (gitURL string, gitRepo string, cle
 
 	labels := map[string]string{"name": gitServerPodName}
 
-	gitServerPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   gitServerPodName,
-			Labels: labels,
-		},
-		Spec: v1.PodSpec{
-			Containers: []v1.Container{
-				{
-					Name:            "git-repo",
-					Image:           imageutils.GetE2EImage(imageutils.Agnhost),
-					Args:            []string{"fake-gitserver"},
-					ImagePullPolicy: "IfNotPresent",
-					Ports: []v1.ContainerPort{
-						{ContainerPort: int32(containerPort)},
-					},
-				},
-			},
-		},
-	}
+	gitServerPod := e2epod.NewAgnhostPod(f.Namespace.Name, gitServerPodName, nil, nil, []v1.ContainerPort{{ContainerPort: int32(containerPort)}}, "fake-gitserver")
```

Comment:
In general, I like the new "e2epod.NewAgnhostPod" function but my vote would e to change the name.
There are some places in this PR where there is no relation to agnhost and it makes the e2e test feel weird.
For example here: a "gitServerPod" is being instantiated through a function called creating an Agnhost (not completely obvious why).

I like the structure of the NewAgnhostPod but I would opt for changing its name to something more generic, if used in places like this one.

Reply:
What would you change the name to?

IMO, ``NewAgnhostPod`` is pretty self-explanatory, it creates a pod using the agnhost image, which is commonly used for most Conformance tests.

For the past few releases, we've centralized 20+ images into agnhost, as there were way too many images to handle, manage, track (especially from my point of view, since I would have to add, build, and maintain all the Windows-equivalent images). One of those centralized images was ``fakegitserver``, which was previously used here. The variable name was kept as ``gitServerPod`` in order to keep code churn at a minimum, making it easier to review, and avoiding too many merge conflicts.

Reply:
@alejandrox1 the agnhost im,age runs a binary that takes a range of commands and fake-gitserver is one of those commands. 

This does make sense after reading https://github.com/kubernetes/kubernetes/tree/master/test/images/agnhost#agnhost 

I think it might be a good idea to consider renaming the args parameter on NewAgnhostPod to agnHostCommand so that as to make it clear that a agnhost command is being chosen here. 

In addition, would be useful somewhere in this file to reference the[agnhost readme](https://github.com/kubernetes/kubernetes/tree/master/test/images/agnhost#agnhost ). 

I think a "see also " reference on the NewAgnhostPod func definition comment would be as good a place as any.